Src/AutoFixture/Dsl/NodeComposer.cs

```diff
@@ -304,6 +304,37 @@ public IPostprocessComposer<T> OmitAutoProperties()
                 when: targetToDecorate.Equals);
         }
 
+        public IPostprocessComposer<T> With<TProperty>(Expression<Func<T, TProperty>> propertyPicker, Func<IFixture, TProperty> generator)
```

Comment:
@zvirja I assume this is the place where most of the work should go. I hope I got it right :)

Reply:
I haven't tried to deeply dig into this method implementation (as the logic is really crazy here 😅), but consider a few things:
- I'd create a private method with `IPostprocessComposer<T> With<TProperty>(Expression<Func<T, TProperty>> propertyPicker, Func<ISpecimenContext, TProperty> generator)` signature. It should be an exact copy of the `public IPostprocessComposer<T> With<TProperty>(Expression<Func<T, TProperty>> propertyPicker, TProperty value)` method below. The only change is that you should pass callback to the `BindingCommand` constructor, rather than a scalar value.
- All the other `With()` implementations should delegate to it and either resolve `TInput` argument, or just return the fixed result.
- Don't activate any instances beforehand and rather activate them during the actual request (inside the callback you pass to the `BindingCommand`). That is because code might be invoked for a few times and you want to have different values for each time.